yarn-project/merkle-tree/src/tree_base_static_initializable.ts

```diff
@@ -0,0 +1,45 @@
+import { LevelUp } from 'levelup';
+import { TreeBase, decodeMeta } from './tree_base.js';
+import { Hasher } from './hasher.js';
+
+export abstract class TreeBaseStaticInitializable extends TreeBase {
```

Comment:
I introduced the `TreeBaseStaticInitializable` because the `StandardIndexedTree` also has these methods but they have sligthly different params and TS doesn't allow you to override methods while having different params (which sounds like a good thing).

> Also, instead of using Reflect.construct, can you use this pattern:
Sure, it looks less complicated.

Reply:
Now that I am thinking about I could unite the functions with IndexedTree. It would just require not allowing for setting the `prefilledSize` but that's probably ok.

Reply:
Agree on removing this class, the two generic factory methods should do the trick. Note that if initialization params don't match exactly, you can also make them generic args of the function, something like this:
```ts
class Base {
  constructor() {}
}
class Foo extends Base {
  constructor(name: string, opts: { a: number }) {
    super();
  }
}

class Bar extends Base {
  constructor(name: string, opts: { b: number }) {
    super();
  }
}

function create<T extends Base, Opt extends object>(c: { new (name: string, opts: Opt): T }, name: string, opts: Opt): T {
  return new c(name, opts);
}

function test() {
  create(Foo, 'foo', { a: 10 });
  create(Bar, 'bar', { b: 10 });
}
```

Or you can just pass in `prefilledSize` anyway and just ignore it when it's not relevant, that's probably the easiest way.

Reply:
@spalladino I just left there `prefilledSize`. It's not the most elegant approach but I think it's ok given that the code will eventually get replaces.